src/coreclr/jit/assertionprop.cpp

```diff
@@ -5722,7 +5722,25 @@ bool Compiler::optCreateJumpTableImpliedAssertions(BasicBlock* switchBb)
             // TODO-Cleanup: We shouldn't attach assertions to nodes in Global Assertion Prop.
             // It limits the ability to create multiple assertions for the same node.
             GenTree* tree = gtNewNothingNode();
-            fgInsertStmtAtBeg(target, fgNewStmtFromTree(tree));
+
+            // Skip all PHI def statements
+            Statement* curStmt = target->firstStmt();
+            while ((curStmt != nullptr) && curStmt->IsPhiDefnStmt())
+            {
+                curStmt = curStmt->GetNextStmt();
+            }
+
+            if (curStmt == nullptr)
+            {
+                // The block is either empty or contains only PHI stmt(s).
+                fgInsertStmtAtEnd(target, fgNewStmtFromTree(tree));
+            }
+            else
+            {
+                // Insert the NOP before the first non-PHI stmt.
+                fgInsertStmtBefore(target, curStmt, fgNewStmtFromTree(tree));
+            }
```

Comment:
`fgInsertStmtAtBeg` already inserts after the first phi node, so this is probably unnecessary.

Reply:
I was addressing Andy's feedback, but looks like indeed it was correct as is, thanks!